tests/unit/Core/Framework/Gateway/Context/Command/ContextGatewayCommandCollectionTest.php

```diff
@@ -0,0 +1,67 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Core\Framework\Gateway\Context\Command;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Gateway\Context\Command\AbstractContextGatewayCommand;
+use Shopware\Core\Framework\Gateway\Context\Command\ChangePaymentMethodCommand;
+use Shopware\Core\Framework\Gateway\Context\Command\ContextGatewayCommandCollection;
+use Shopware\Core\Framework\Gateway\Context\Command\LoginCustomerCommand;
+use Shopware\Core\Framework\Gateway\Context\Command\RegisterCustomerCommand;
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+#[CoversClass(ContextGatewayCommandCollection::class)]
+class ContextGatewayCommandCollectionTest extends TestCase
+{
+    public function testGetTokenCommandWithRegister(): void
+    {
+        $commands = new ContextGatewayCommandCollection();
+        $commands->add(self::getCommand(RegisterCustomerCommand::class, ['data' => ['foo' => 'bar']]));
+        $commands->add(self::getCommand(RegisterCustomerCommand::class, ['data' => ['wow' => 'ser']]));
+
+        /** @var RegisterCustomerCommand $registerCommand */
```

Comment:
please do not use var annotations. Especially in tests you can simply assert the type. also see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md